### ISSUE TITLE
[WebAPI] Hide fragment directives from Navigation Timing API

### ISSUE THREAD

**Issue by hamishwillee** [mdn/content#36546]
## Acceptance criteria

- [x] The listed features are documented sufficiently on MDN
- [x] BCD is updated
- [x] Interactive example and data repos are updated if appropriate
- [x] The content has been reviewed as needed

### For folks helping with Firefox related documentation

- [x] Set bugs to `dev-doc-complete`
- [x] Add entry to Firefox release notes for enabled/preview features
- [ ] Add/remove entry to Firefox experimental features page for preference/released features

## Features to document

This needs at least a release note because the behaviour is a bugfix that unfortunately some sites are relying on. 
I would say a BCD entry too.

It may be reasonable to add a note to the preformance API thingy too that this is URL minus any url fragment (unrelated, but same behaviour might b recorded in https://developer.mozilla.org/en-US/docs/Web/API/Location/href

What happens is that if you use [`Performance.getEntries()`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/getEntries) (or the variants that take a name and type) you get an array of entries that have a particular requested type. One of the entry types is [navigation](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/entryType#navigation) so your objects are [PerformanceNavigationTiming](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming).

For these the `name` of each entry is a URL. The change is that now this URL will exclude the text fragment.

## Related Gecko bugs

https://bugzilla.mozilla.org/show_bug.cgi?id=1919565

## Other

- [x] Check content open issues to see if any pertain to the subject matter. If there are any that can be closed because of the work, do so. If there are any that can be fixed relatively quickly because of the knowledge from completing this issue and you have time, feel free to go ahead and fix them.

**Comment by eligrey** [comment]
> the behaviour is a bugfix that unfortunately some sites are relying on.

That is incorrect. This was not a bugfix prior to Firefox supporting scroll-to-text. Transcend shipped this feature long before Firefox shipped scroll-to-text support.

I reached out to @zcorpan and have not received proper engagement for my concerns. I believe it is unjust for Mozilla to retroactively classify breaking Transcend's features as a 'bugfix', given that there was no 'bug' when we first added the feature.
